commands/connect/mapping-write-errors.js

```diff
@@ -1,37 +1,36 @@
 'use strict'
 const api = require('../../lib/connect/api.js')
 const regions = require('../../lib/connect/regions.js')
+const pg = require('heroku-pg')
 const cli = require('heroku-cli-util')
 const co = require('co')
 
 function * run (context, heroku) {
   context.region = yield regions.determineRegion(context, heroku)
-  let mappingName = context.args.name
+  let mappingName = context.args.name.toLowerCase()
   let connection = yield api.withConnection(context, heroku)
-  let mapping = yield api.withMapping(connection, mappingName)
-  let results = yield cli.action('Retrieving errors', co(function * () {
-    let url = '/api/v3/mappings/' + mapping.id + '/errors'
-    return yield api.request(context, 'GET', url)
-  }))
-  let errors = results.json
+  let query = `
+SELECT
+  id, processed_at, record_id, sf_message, false AS archived
+FROM
+  ${connection.schema_name}._trigger_log
+WHERE
+  state = 'FAILED' AND table_name = '${mappingName}'
+UNION
+SELECT
+  id, processed_at, record_id, sf_message, true AS archived
+FROM
+  ${connection.schema_name}._trigger_log_archive
+WHERE
+  state = 'FAILED' AND table_name = '${mappingName}'
+ORDER BY
+  processed_at DESC
```

Comment:
I would limit the output, the sorting can be CPU intense for lager volumes of data. There are no default statement timeouts configured on the Postgres instances. So a limit would be a sanity check, just in case ;)